blog/2022/05-10-resoto-graph-traversal/index.md

```diff
@@ -0,0 +1,119 @@
+---
+authors: [matthias]
+tags: [graph, search, digitalocean]
+---
+
+# A walk in the graph
+
+Resoto uses a [directed graph](https://en.wikipedia.org/wiki/Directed_graph) to represent your infrastructure resources as nodes and relationships between them as edges. It stores that resource data in a graph database, allowing you to search the graph efficiently. Why should I care, you ask? To answer this question, I would like to present an example that demonstrates the power of this approach.
+
+The following diagram shows an excerpt of the graph model for DigitalOcean droplets. Droplets are the compute instances in DigitalOcean, similar to EC2 instances in AWS. The full model is available in our [reference section](/docs/reference/data-models/digitalocean). Please note that this example uses DigitalOcean, but the same also works on [AWS](/docs/reference/data-models/aws) or [Google Cloud](/docs/reference/data-models/gcp).
+
+![DigitalOcean Droplet](img/digitalocean_droplet.svg)
+
+This diagram shows how the different kinds of resources are connected to each other in the graph. This does not mean that all of these resources have to exist in your infrastructure. It only shows the structure of a possible graph. A droplet, for example, can have volumes, snapshots, or floating IPs, which are direct successors of the node. If we want to know which volumes are attached to a droplet, we will walk the graph outbound by one step. Since this could yield volumes, snapshots, and floating IPs, as seen in the diagram, we need to filter the resulting list to only keep the volumes. Let's assume we have a specific droplet with id `289061882`, where we want to know all attached volumes:
```

Comment:
"this does not mean that all of these resources have to exist" --> not sure what that means? do they or do they not? 

I think in this paragraph you want to first explain the concept of how a single resource may have successors and ancestors, and why that's the case. You probably want to refer to Infrastructure-as-code in that explanation. 

Split up the paragraph as well. It's too dense to read.